src/ir/ty.rs

```diff
@@ -143,17 +127,15 @@ impl Type {
             TypeKind::BlockPointer |
             TypeKind::Int(..) |
             TypeKind::Float(..) |
-            TypeKind::Named(..) => true,
+            TypeKind::Named => true,
             _ => false,
         }
     }
 
     /// Creates a new named type, with name `name`.
     pub fn named(name: String) -> Self {
         assert!(!name.is_empty());
```

Comment:
Well, the point is that invalid identifiers may come here, we just don't want to generate code for them. It'd be good if they didn't, but more work needs to happen so that's true.